src/OpenRiaServices.Server.EntityFrameworkCore/Framework/DbContextEFCoreExtensions.cs

```diff
@@ -109,17 +109,24 @@ private static void AttachAsModifiedInternal<T>(EntityEntry<T> stateEntry, T ori
 
             foreach (var member in stateEntry.CurrentValues.Properties)
             {
-                stateEntry.OriginalValues[member] = originalValues[member];
-
-                // For any members that don't have RoundtripOriginal applied, EF can't determine modification
-                // state by doing value comparisons. To avoid losing updates in these cases, we must explicitly
-                // mark such members as modified.
                 PropertyDescriptor property = properties[member.Name];
-                if (property != null &&
-                    (!isRoundtripType && property.Attributes[typeof(RoundtripOriginalAttribute)] == null) &&
-                    property.Attributes[typeof(ExcludeAttribute)] == null)
+
+                if (property != null)
```

Comment:
Should we add a comment about this protecting against ShadowProperties (and maybe exclude attributes ?)

Reply:
Added comment